File: modules/core/src/main/java/org/apache/ignite/internal/processors/platform/websession/SessionStateLockEntryResult.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.ignite.internal.processors.platform.websession;

import org.apache.ignite.binary.BinaryObjectException;
import org.apache.ignite.binary.BinaryRawReader;
import org.apache.ignite.binary.BinaryRawWriter;
import org.apache.ignite.binary.BinaryReader;
import org.apache.ignite.binary.BinaryWriter;
import org.apache.ignite.binary.Binarylizable;
import org.apache.ignite.internal.util.typedef.internal.S;

import java.sql.Timestamp;

/**
 * Result of the {@link LockEntryProcessor} execution.
 */
@SuppressWarnings({"AssignmentToDateFieldFromParameter", "ReturnOfDateField"})
public class SessionStateLockEntryResult implements Binarylizable {
    /** Success flag. */
    private boolean success;

    /** Data. */
    private SessionStateData data;

    /** Lock time. */
    private Timestamp lockTime;

    /**
     * Ctor.
     *
     * @param success Success flag.
     * @param data Session data.
     * @param lockTime Lock time.
     */
    public SessionStateLockEntryResult(boolean success, SessionStateData data, Timestamp lockTime) {
        this.success = success;
        this.data = data;
        this.lockTime = lockTime;
    }

    /**
     * @return Success flag.
     */
    public boolean success() {
        return success;
    }

    /**
     * @return Session state data.
     */
    public SessionStateData data() {
        return data;
    }

    /**
     * @return Lock time.
     */
    public Timestamp lockTime() {
        return lockTime;
    }

    /** {@inheritDoc} */
    @Override public void writeBinary(BinaryWriter writer) throws BinaryObjectException {
        BinaryRawWriter raw = writer.rawWriter();

        raw.writeBoolean(success);
        raw.writeObject(data);
        raw.writeTimestamp(lockTime);
    }

    /** {@inheritDoc} */
    @Override public void readBinary(BinaryReader reader) throws BinaryObjectException {
        BinaryRawReader raw = reader.rawReader();

        success = raw.readBoolean();
        data = raw.readObject();
        lockTime = raw.readTimestamp();
    }

    /** {@inheritDoc} */
    @Override public String toString() {
        return S.toString(SessionStateLockEntryResult.class, this);
    }
}

File: modules/platforms/dotnet/Apache.Ignite.Core/Impl/AspNet/SessionStateLockEntryResult.cs
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

namespace Apache.Ignite.Core.Impl.AspNet
{
    using System;
    using System.Diagnostics;
    using Apache.Ignite.Core.Binary;

    /// <summary>
    /// Result of the session state lock processor.
    /// </summary>
    public class SessionStateLockEntryResult
    {
        /** Success flag. */
        private readonly bool _success;

        /** Session state data. */
        private readonly SessionStateData _data;

        /** Lock time. */
        private readonly DateTime? _lockTime;

        /// <summary>
        /// Initializes a new instance of the <see cref="SessionStateLockEntryResult"/> class.
        /// </summary>
        /// <param name="reader">The reader.</param>
        public SessionStateLockEntryResult(IBinaryRawReader reader)
        {
            _success = reader.ReadBoolean();
            _data = reader.ReadObject<SessionStateData>();
            _lockTime = reader.ReadTimestamp();

            Debug.Assert(_success ^ (_data == null));
            Debug.Assert(_success ^ (_lockTime != null));
        }

        /// <summary>
        /// Gets a value indicating whether lock succeeded.
        /// </summary>
        public bool Success
        {
            get { return _success; }
        }

        /// <summary>
        /// Gets the data. Null when <see cref="Success"/> is <c>false</c>.
        /// </summary>
        public SessionStateData Data
        {
            get { return _data; }
        }

        /// <summary>
        /// Gets the lock time. Null when <see cref="Success"/> is <c>true</c>.
        /// </summary>
        public DateTime? LockTime
        {
            get { return _lockTime; }
        }
    }
}
